.changeset/orange-moles-leave.md

```diff
@@ -0,0 +1,5 @@
+---
+"@fuel-ts/abi-coder": patch
+---
+
+fix: handling very large numbers gracefully
```

Comment:
Good stuff - some [optional] nits below.

Since you are throwing, this is not necessarily graceful, nor should it be.

I'd change the title to something like:
 - `fix: validation and handling of unsafe integers`

And the release note to something like:
 - Improved validation and handling of unsafe integers in `BigNumberCoder`